### PR TITLE
Add jenkinsfile template

### PR DESCRIPTION
**Pull request by RudolfDG**

### CODE REVIEW

File: templates/jenkins/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+                    script {
+                        sh '''#!/bin/bash
+                        login_oc.sh $OC_URL > /dev/null || echo ok
+                        oc project $OC_PROJECT
+                        oc tag $APP_NAME:$GIT_SHORT_COMMIT $APP_NAME:qas
+                        # Patch the deployment in order to trigger rollout
+                        # oc patch deployment $APP_NAME-qas -p '{"spec": {"template": {"metadata": {"annotations": {"git-ref": "$IMAGE_TAG"}}}}}'
+                        # Check the status of the rollout
+                        oc rollout status deployment/$APP_NAME-qas --watch=true
+                        '''
```
Comment (spacid) [inline templates/jenkins/Jenkinsfile]: Can be changed to `tagNewImage('qas')` ?

File: templates/jenkins/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+void getImageFromDockerfile() {
+    return 'python:3.7'
+}
+
+void setGitHubBuildStatus(String message, String state) {
+    step([
+        $class: 'GitHubCommitStatusSetter',
+        reposSource: [$class: 'ManuallyEnteredRepositorySource', url: "${GIT_URL}"],
+        commitShaSource: [$class: 'ManuallyEnteredShaSource', sha: "${GIT_COMMIT}"],
+        //contextSource: [$class: "ManuallyEnteredCommitContextSource", context: "ci/jenkins/build-status"],
```
Comment (spacid) [inline templates/jenkins/Jenkinsfile:176]: Let's remove code that is commented out.

File: templates/jenkins/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        stage('Deploy INT') {
+            when {
+                anyOf {
+                    changeRequest target: 'master'
+                    changeRequest target: 'main'
+                }
+            }
+            steps {
+                container('oc') {
+                    tagNewImage('qas')
```
Comment (spacid) [inline templates/jenkins/Jenkinsfile]: should be `tagNewImage('int')`?

File: templates/jenkins/Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        }
+        stage('Build code') {
+            when {
+                not {
+                    buildingTag()
+                }
+            }
+            steps {
+                container('oc') {
+                    tagNewImage('int')
```
Comment (spacid) [inline templates/jenkins/Jenkinsfile]: Always tag/rollout int when building code? Probably copy/paste issue as the build code has been replaced.

**Review by spacid — CHANGES_REQUESTED** [review]

**Review by spacid — APPROVED** [review]

**Review by violetina — APPROVED** [review]
looks good

**Pull request merged**